File: 프로그래머스/3/42898. 등굣길/등굣길.py
def solution(m, n, puddles):
    answer = 0
    dp = [[0]*(n+1) for _ in range(m+1)]
    
    for r in range(1, n+1):
        for c in range(1, m+1):
            
            if [c, r] in puddles:
                continue
                
            if c==1 and r==1:
                dp[1][1] = 1
                continue

            dp[c][r] = dp[c-1][r]+ dp[c][r-1]
    return dp[m][n] % 1000000007

File: 프로그래머스/3/42898. 등굣길/등굣길.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(int m, int n, vector<vector<int>> puddles) {
    int answer = 0;
    vector<vector<long long>> board(n+1, vector<long long>(m+1,0));
    
    for(int r=1; r<=n; r++){
        for(int c=1; c<=m; c++){
            if(r==1 && c==1) board[r][c]=1;
            else if(find(puddles.begin(), puddles.end(), vector<int>{c, r})!=puddles.end()) continue;
            else{
                board[r][c] = (board[r-1][c]+board[r][c-1])%1000000007;
            }
        }
    }
    answer = board[n][m];
    return answer;
}